src/partisan/irods.py

```diff
@@ -1046,11 +1045,38 @@ def __str__(self):
         )
 
 
+def rods_type_check(method):
+    """Adds a check to RodsItem methods that ensures the item's path in iRODS has the
+    appropriate type. i.e. that a Collection has a collection path and a DataObject has
+    a data object path."""
+
+    @wraps(method)
+    def wrapper(*args, **kwargs):
+        item = args[0]
+        if item.check_type and not item.type_checked:
+            item.check_rods_type(**kwargs)
+            item.type_checked = True
+            return method(*args, **kwargs)
+        else:
+            return method(*args, **kwargs)
+
+    return wrapper
+
+
 class RodsItem(PathLike):
     """A base class for iRODS path entities."""
 
-    def __init__(self, path: Union[PurePath, str], pool=default_pool):
+    def __init__(self, path: Union[PurePath, str], check_type=False, pool=default_pool):
+        """
+        RodsItem constructor.
+        Args:
+            path: A remote path.
+            check_type: Check the remote path type if True, defaults to False.
+            pool: A baton client pool. Optional.
+        """
         self.path = PurePath(path)
```

Comment:
Why is the abstract class PurePath used? Is that related to some later features compatible to other operative systems?

Reply:
`PurePath` is not abstract in the object-oriented programming sense. "Pure path objects provide path-handling operations which don’t actually access a filesystem." (see https://docs.python.org/3/library/pathlib.html). This makes them perfect for representing iRODS paths, which are virtual. A data object has a path stored in the iRODS database which is not the same as the path(s) to the files(s) holding the data.